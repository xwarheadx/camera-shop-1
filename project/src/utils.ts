import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


