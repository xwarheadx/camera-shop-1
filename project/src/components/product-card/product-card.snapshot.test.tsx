import { productMock } from '../../test/test-mocks';
import { renderWithProviders } from '../../test/utils/render-with-redux';
import ProductCard from './product-card';

test('Render ProductCard', () => {
  const { asFragment } = renderWithProviders(<ProductCard product={productMock}/>);
  expect(asFragment()).toMatchSnapshot();
});
