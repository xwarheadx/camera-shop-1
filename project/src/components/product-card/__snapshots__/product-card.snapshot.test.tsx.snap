// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProductCard 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product-card-test"
  >
    <div
      class="product-card__img"
    >
      <picture>
        <source
          srcset="../img/content/look-identify.webp, ../img/content/look-identify@2x.webp 2x"
          type="image/webp"
        />
        <img
          alt="Look Identify"
          height="240"
          src="../img/content/look-identify.jpg"
          srcset="../img/content/look-identify@2x.jpg 2x"
          width="280"
        />
      </picture>
    </div>
    <div
      class="product-card__info"
    >
      <div
        class="rate product-card__rate"
      >
        <svg
          aria-hidden="false"
          height="16"
          width="17"
        >
          <use
            data-testid="star-test"
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="false"
          height="16"
          width="17"
        >
          <use
            data-testid="star-test"
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="false"
          height="16"
          width="17"
        >
          <use
            data-testid="star-test"
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="false"
          height="16"
          width="17"
        >
          <use
            data-testid="star-test"
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="false"
          height="16"
          width="17"
        >
          <use
            data-testid="star-test"
            xlink:href="#icon-full-star"
          />
        </svg>
        <p
          class="visually-hidden"
          data-testid="rating-test"
        >
          Рейтинг: 5
        </p>
        <p
          class="rate__count"
        >
          <span
            class="visually-hidden"
            data-testid="rates-total"
          >
            Всего оценок:
          </span>
          18
        </p>
      </div>
      <p
        class="product-card__title"
      >
        Look Identify
      </p>
      <p
        class="product-card__price"
        data-testid="price-test"
      >
        <span
          class="visually-hidden"
        >
          Цена:
        </span>
        12 6000 ₽
      </p>
    </div>
    <div
      class="product-card__buttons"
    >
      <button
        class="btn btn--purple product-card__btn"
        type="button"
      >
        Купить
      </button>
      <a
        class="btn btn--transparent"
        data-testid="link-test"
        href="/cameras/17"
      >
        Подробнее
      </a>
    </div>
  </div>
</DocumentFragment>
`;
